opencxl/cxl/component/cache_route_table_manager.py

```diff
@@ -0,0 +1,70 @@
+"""
+ Copyright (c) 2024, Eeum, Inc.
+
+ This software is licensed under the terms of the Revised BSD License.
+ See LICENSE for details.
+"""
+
+from dataclasses import dataclass
+from typing import TypedDict, List
+
+from opencxl.util.component import LabeledComponent
+from opencxl.util.logger import logger
+
+
+class CacheRouteTableCapabilities(TypedDict):
+    # pylint: disable=duplicate-code
+    cache_id_target_count: int
+    hdmd_type2_device_max_count: int
+    explicit_cache_id_rt_cmt_required: int
+
+
+@dataclass
+class CacheRouteTableBase(LabeledComponent):
+    _capabilities: CacheRouteTableCapabilities
+    _target_count: int
+    _cache_id_to_port_mapping: List[int]
+    _port_to_cache_id_mapping: dict[int, int]
+
+    def __init__(self, capabilities, label):
+        super().__init__(label)
+        self._capabilities = capabilities
+        self._target_count = capabilities["cache_id_target_count"]
+        self._cache_id_to_port_mapping = [0] * self._target_count
+        self._port_to_cache_id_mapping = {}
+
+    def get_target(self, cache_id) -> int:
+        return self._cache_id_to_port_mapping[cache_id]
+
+    def get_cache_id(self, target) -> int:
+        return self._port_to_cache_id_mapping[target]
+
+
+@dataclass
+class SwitchCacheRouteTable(CacheRouteTableBase):
+    """
+    For a CXL Switch, we have 16 cache route table entries (for 256B flit mode).
+    """
+
+    def __init__(self, capabilities, label):
+        if capabilities["cache_id_target_count"] != 0x10:
+            raise ValueError(
+                "CXL switch routing table must have 16 cache route table entries "
+                f"but {capabilities.cache_id_target_count} were given"
+            )  # TODO: support 68B flit mode?
+        super().__init__(capabilities, label)
+
+    # TODO: properly implement commits
```

Comment:
Why not implement on this PR?

Reply:
This is the bare minimum to get cache routing working wrt the demo app. What I can do is just apply the two commits as is onto my demo app branch, and add in more advanced commit functionality after we're done with the demo.

Reply:
OK, we can push it to v0.3-dev after you address the other comments. Be sure to not close that ticket so that we can track this as a remaining task.